src/scanner.c

```diff
@@ -84,12 +84,12 @@ enum TokenType {
     HEREDOC_IDENTIFIER,
 };
 
-enum ContextType {
+typedef enum ContextType {
     TEMPLATE_INTERPOLATION,
     TEMPLATE_DIRECTIVE,
     QUOTED_TEMPLATE,
     HEREDOC_TEMPLATE,
-};
+} ContextType;
```

Comment:
You're missing these changes in the terraform scanner.

Reply:
:see_no_evil: that explains it; sorry and thank you! Is there a way to handle the terraform scanner more elegantly? I put it there because at the time nvim-treesitter didnt allow having queries for multiple filetypes with one parser.

Reply:
If it's really the same parser, you can register it for other filetypes with `vim.treesitter.language.register()`.

Reply:
Queries differ somewhat ( there used to be only one, but then there were some complaints that some strings got highlighted as terraform keywords in other hcl based languages like for packer or whatnot, so i split it into generic hcl and then terraform iirc )

Reply:
Then I think this is the best for now (until upstream tree-sitter adds dedicated "dialect" support).